foundry-voting/circuits/src/main.nr

```diff
@@ -1,87 +1,53 @@
-use dep::std;
 
-fn main(
-    root: pub Field,
-    index: Field,
-    hash_path: [Field; 2],
-    secret: Field,
-    proposalId: pub Field,
-    vote: pub Field
-) -> pub Field {
+fn main(root: pub Field, index: Field, hash_path: [Field;2], secret: Field, proposalId: pub Field, vote: pub u8) -> pub Field {
     let note_commitment = std::hash::pedersen_hash([secret]);
     let nullifier = std::hash::pedersen_hash([root, secret, proposalId]);
 
-    let check_root = std::merkle::compute_merkle_root(note_commitment, index, hash_path);
-    assert(root == check_root);
-
-    // Originally contrained the vote to avoid front-running,
-    // but including the vote as a public input is sufficient
-
-    // assert(vote <= 1);
+    let merkle_root = std::merkle::compute_merkle_root(note_commitment, index, hash_path);
```

Comment:
If you want, you can replace the MT with the MTs from [zk-kit](https://github.com/privacy-scaling-explorations/zk-kit.noir), this would be a cool example of how they're used

Reply:
Just checkout out this package zk-kit, its really cool, thanks for sharing it. yes I'll add it here as well.

Reply:
Done. Now we using MT from `zk-kit`.